### ISSUE TITLE
`Error: nuxt instance unavailable` when signing out from a protected page

### ISSUE THREAD

**Issue by LLCampos** [sidebase/nuxt-auth#132]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.13.2`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@9.2.0`
- Builder:          `vite`
- User Config:      `css`, `modules`, `auth`
- Runtime Modules:  `@nuxtjs/tailwindcss@6.1.0`, `@sidebase/nuxt-auth@0.3.2`

### Reproduction

https://github.com/LLCampos/nuxt-auth-example

This [commit](https://github.com/LLCampos/nuxt-auth-example/commit/fd1d0118e07127e3a7fd51c1269cc6eb08e6e70a) contains the changes necessary to the current example app to reproduce the issue.

### Describe the bug

* Start application
* It should immediately take you to login page
* Login with credentials
* Logout

I get the following in logs

```
[Vue Router warn]: uncaught error during route navigation:                                                                                                            11:09:23
Error: nuxt instance unavailable                                                                                                                                      11:09:23
    at useNuxtApp (/Users/luiscampos/Development/nuxt-auth-example/node_modules/nuxt/dist/app/nuxt.mjs:165:13)
    at Module.useRuntimeConfig (/Users/luiscampos/Development/nuxt-auth-example/node_modules/nuxt/dist/app/nuxt.mjs:173:10)
    at getApiURL (/Users/luiscampos/Development/nuxt-auth-example/node_modules/@sidebase/nuxt-auth/dist/runtime/utils/url.mjs:10:40)
    at Module.joinPathToApiURL (/Users/luiscampos/Development/nuxt-auth-example/node_modules/@sidebase/nuxt-auth/dist/runtime/utils/url.mjs:15:66)
    at /Users/luiscampos/Development/nuxt-auth-example/node_modules/@sidebase/nuxt-auth/dist/runtime/middleware/auth.mjs:16:65
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.callAsync (file:///Users/luiscampos/Development/nuxt-auth-example/node_modules/unctx/dist/index.mjs:53:16)
    at async /Users/luiscampos/Development/nuxt-auth-example/node_modules/nuxt/dist/pages/runtime/router.mjs:110:22
[Vue Router warn]: uncaught error during route navigation:                                                                                                            11:09:23
Error: nuxt instance unavailable                                                                                                                                      11:09:23
    at useNuxtApp (/Users/luiscampos/Development/nuxt-auth-example/node_modules/nuxt/dist/app/nuxt.mjs:165:13)
    at Module.useRuntimeConfig (/Users/luiscampos/Development/nuxt-auth-example/node_modules/nuxt/dist/app/nuxt.mjs:173:10)
    at getApiURL (/Users/luiscampos/Development/nuxt-auth-example/node_modules/@sidebase/nuxt-auth/dist/runtime/utils/url.mjs:10:40)
    at Module.joinPathToApiURL (/Users/luiscampos/Development/nuxt-auth-example/node_modules/@sidebase/nuxt-auth/dist/runtime/utils/url.mjs:15:66)
    at /Users/luiscampos/Development/nuxt-auth-example/node_modules/@sidebase/nuxt-auth/dist/runtime/middleware/auth.mjs:16:65
    at async Object.callAsync (file:///Users/luiscampos/Development/nuxt-auth-example/node_modules/unctx/dist/index.mjs:53:16)
    at async /Users/luiscampos/Development/nuxt-auth-example/node_modules/nuxt/dist/pages/runtime/router.mjs:110:22
[nitro] [dev] [unhandledRejection] Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client                                                11:09:23
    at new NodeError (node:internal/errors:371:5)
    at ServerResponse.setHeader (node:_http_outgoing:576:11)
    at setResponseHeader (file:///Users/luiscampos/Development/nuxt-auth-example/node_modules/nitropack/node_modules/h3/dist/index.mjs:277:18)
    at Object.errorhandler [as onError] (file:///Users/luiscampos/Development/nuxt-auth-example/.nuxt/dev/index.mjs:450:5)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Server.toNodeHandle (file:///Users/luiscampos/Development/nuxt-auth-example/node_modules/nitropack/node_modules/h3/dist/index.mjs:752:9) {
  code: 'ERR_HTTP_HEADERS_SENT'
```

**Comment by BracketJohn** [comment]
Wow, you have been busy @LLCampos! 🏆  Will look into this ASAP, thanks! It's very likely related to nasty, nested context behavior -> something very deeply engrained in Vue 3 / Nuxt 3. See https://github.com/nuxt/framework/issues/5740#issuecomment-1229197529

**Comment by BracketJohn** [comment]
Brief update on this front: We have a fix in #136 -> i basically had to reverse-engineer a lot of how nuxt and `vue-router` handles `navigateTo` and the fix still has to pass reviiew of @JoaoPedroAS51, but afterwards we should be good to go (:
